### PR TITLE
Avoid reading state from register on startup

### PR DESCRIPTION
**Pull request by DaveCTurner**
Now that #94497 ensures we read the state from the register during the election-winning process, there's no need to read it at startup. With this commit we start with an empty state every time.

### CODE REVIEW

**Comment by DaveCTurner** [comment]
This seems plausible but unfortunately doesn't quite work yet: I get (very) occasional failures in stabilisation even with the longer timeout. I will dig into them more deeply tomorrow.

**Comment by DaveCTurner** [comment]
At least one of the failures was (largely) unrelated and I expect https://github.com/elastic/elasticsearch/pull/94726 will fix it.

File: server/src/test/java/org/elasticsearch/cluster/coordination/AtomicRegisterCoordinatorTests.java [hunk 1]
```diff
@@ -183,6 +183,17 @@ public void testCannotJoinClusterWithDifferentUUID() {
...
 
+    @Override
+    @AwaitsFix(bugUrl = "ES-5645")
+    public void testClusterUUIDLogging() {
+        // Stateless masters do not lock in to a cluster UUID in a way that persists across restarts so this test doesn't make sense here
+    }
+
+    @Override
+    protected long extraStabilisationTime() {
+        return 300000; // TODO liveness improvements
```
Comment (fcofdez) [inline server/src/test/java/org/elasticsearch/cluster/coordination/AtomicRegisterCoordinatorTests.java:194]: Is this still needed with the fix from https://github.com/elastic/elasticsearch/pull/94726?

Comment (DaveCTurner) [reply server/src/test/java/org/elasticsearch/cluster/coordination/AtomicRegisterCoordinatorTests.java:194]: Possibly not. I expect we will  need to allow for _some_ extra time for stabilisation in these tests, especially if we make some of the possibly-async steps into genuinely-async ones. But this wasn't that, it's just the first thing I try when looking at these tests.

**Review by fcofdez — APPROVED** [review]
LGTM, nice change :)

**Pull request merged**